src/System.Private.CoreLib/shared/System/Collections/Generic/Dictionary.cs

```diff
@@ -968,7 +974,9 @@ public int EnsureCapacity(int capacity)
         {
             if (capacity < 0)
                 ThrowHelper.ThrowArgumentOutOfRangeException(ExceptionArgument.capacity);
-            int currentCapacity = _entries == null ? 0 : _entries.Length;
+
+            // _entries.Length == 1 is dummy entries
+            int currentCapacity = _entries.Length == 1 ? 0 : _entries.Length;
```

Comment:
So it's removing a null check but adding in one or more _entries.Length == 1 checks?

Reply:
The idea was remove check null for main method that adds so here we don't have null anymore but dummy len. BTW I agree with @jkotas and this improvement is not good for Dictionary, but as I said I published my thought/result

Reply:
https://github.com/dotnet/coreclr/pull/22599#issuecomment-463694115 do you agree?

Reply:
> do you agree?

Yes.  Thanks.

Reply:
I'm exploring if some improvement done on DictionartSlim(corefxlab) can improve Dictionary.